src/pygeoprocessing/geoprocessing.py

```diff
@@ -2267,10 +2269,16 @@ def calculate_disjoint_polygon_set(
     for poly_feat in vector_layer:
         poly_geom_ref = poly_feat.GetGeometryRef()
         if poly_geom_ref is None:
-            LOGGER.warn(
+            LOGGER.warning(
                 f'no geometry in {vector_path} FID: {poly_feat.GetFID()}, '
                 'skipping...')
             continue
+        if poly_geom_ref.IsEmpty():
+            LOGGER.warning(
+                f'empty geometry in {vector_path} FID: {poly_feat.GetFID()}, '
+                'skipping...')
+            continue
+        LOGGER.info(poly_geom_ref.ExportToWkt())
```

Comment:
Did you mean to leave this in? If so, maybe add a prefix to the logging message so it's clearer where it comes from?
```suggestion
        LOGGER.info('calculate_disjoint_polygon_set: adding polygon ' + poly_geom_ref.ExportToWkt())
```

Reply:
Lines 2265-2267 say
```
# As much as I want this to be in a comprehension, a comprehension version
# of this loop causes python 3.6 to crash on linux in GDAL 2.1.2 (which is
# what's in the debian:stretch repos.)
```
since we're not supporting those versions any more, is it worth refactoring this?

Reply:
> Did you mean to leave this in?

Ooo, yeah I think this'll be better to leave out.  Since logging flushes to a file on each write, this loop would really easily bottleneck with this logger message in there.  Removed in ae0d5ed

Reply:
> since we're not supporting those versions any more, is it worth refactoring this?

I think the loop has enough complexity now where a comprehension, while possible, would be less readable and less maintainable.  We'd have to kind of bend over backwards to shoehorn the same logger.warnings in there.  So if you're OK with it, I think I'd rather just remove the comment and then leave it as a standard for-loop.

Comment removed in 48ed37c